### PR TITLE
Using local notifications and nc-notifications if not found on NETCONF Server

### PR DESCRIPTION
**Pull request by esmasth**
If NETCONF server supports RFC5277 notification capability and
libnetconf2 required notifications and nc-notifications are not present
on the NETCONF server (which it is not obligated to support), then the
libyang context needs to be initialized using client side local YANG
schema files.

This change assumes that notifications and nc-notifications models are
present in searchpath directory set via nc_client_set_schema_searchpath
or in the default YANG_MODULE_DIR set via build time configuration.

This addresses #492

### CODE REVIEW

**Review by michalvasko — APPROVED** [review]

**Review by michalvasko — CHANGES_REQUESTED** [review]

File: src/session_client.c [hunk 1]
```diff
@@ -988,6 +995,26 @@ build_module_info_yl(struct nc_session *session, int get_data_sup, int xpath_sup
         }
     }
 
+    /* If NETCONF server supports RFC5277 notification capability and libnetconf2
+       required notifications and nc-notifications are not present on the NETCONF
+       server (which it is not obligated to support), then the libyang context
+       needs to be initialized using client side local YANG schema files */
```
[the remark this answers is not in the document]
Comment (esmasth) [reply src/session_client.c]: Updated in latest push

File: src/session_client.c [hunk 2]
```diff
@@ -988,6 +995,26 @@ build_module_info_yl(struct nc_session *session, int get_data_sup, int xpath_sup
         }
     }
 
+    /* If NETCONF server supports RFC5277 notification capability and libnetconf2
+     * required notifications and nc-notifications are not present on the NETCONF
+     * server (which it is not obligated to support), then the libyang context
+     * needs to be initialized using client side local YANG schema files */
+    if (nc_session_cpblt(session, "urn:ietf:params:netconf:capability:notification:1.0") &&
+        !notifications_found && !nc_notifications_found) {
```
Comment (michalvasko) [inline src/session_client.c:1003]: Another formatting error, this line needs one more indent.

**Review by michalvasko — CHANGES_REQUESTED** [review]

File: src/session_client.c (hunk 2 above)
Comment (esmasth) [reply src/session_client.c]: Resolved in latest push

**Review by michalvasko — APPROVED** [review]

**Comment by esmasth** [comment]
Thanks @michalvasko!

**Pull request merged**